Assignment-6/src/classes/TriggerHandler_Test.cls

```diff
@@ -0,0 +1,261 @@
+@isTest
+private class TriggerHandler_Test {
+    private static final String TRIGGER_CONTEXT_ERROR = 'Trigger handler called outside of Trigger execution';
+    private static String lastMethodCalled;
+    private static TriggerHandler_Test.TestHandler handler;
+    static {
+        handler = new TriggerHandler_Test.TestHandler();
+        // override its internal trigger detection
+        handler.isTriggerExecuting = true;
+    }
+
+    /***************************************
+    * unit tests
+    ***************************************/
+
+    // contexts tests
+    @isTest
+    static void testBeforeInsert() {
+        beforeInsertMode();
+        handler.run();
+        System.assertEquals('beforeInsert', lastMethodCalled, 'last method should be beforeInsert');
```

Comment:
In the test class use Assert class and its methods instead of using System.assertEquals !!

Reply:
Sir, this is TriggerHandler_test class which was provided to us by Romesh sir for trigger framework. I didn't write it myself.